hook.js

```diff
@@ -39,16 +35,36 @@ function handle(request, response) {
     }
 }
 
-// TODO: Remove the commentURL parameter once parseCSS is refactored
-function addPullRequestComments(destinationRepo, originRepo, originBranch, prNumber, commentURL) {
+function processPullRequest(destinationRepo, originRepo, originBranch, prNumber, commentURL) {
+    var commits = getPullRequestCommits(destinationRepo, prNumber);
     var config = getConfig(originRepo, originBranch);
-    var pullRequestCommits = getPullRequestCommits(destinationRepo, prNumber);
 
-    Q.all([config, pullRequestCommits]).spread(function(config, pullRequestCommits) {
-        pullRequestCommits.forEach(function(currentCommit) {
-            getCommitDetail(originRepo, currentCommit.sha, function(error, currentCommitDetail) {
-                if (error) return logger.error('getCommitDetail failed:', error);
-                parseCSS(currentCommitDetail.files, config, commentURL, token, function(usageInfo) {}, currentCommit.sha);
+    // Once we pull down commit metadata and configuration...
+    Q.all([commits, config]).spread(function(commits, config) {
+
+        // Go through each commit...
+        commits.forEach(function(currentCommit) {
+
+            // And each file of each commit...
+            currentCommit.files.forEach(function(file) {
+
+                // And process the files
+
```

Comment:
I added the extra line just to show that the comment applies to the entire body. I can add it to line 48.